cli/flox-rust-sdk/src/models/env_registry.rs

```diff
@@ -118,8 +121,27 @@ impl EnvRegistry {
     }
 
     /// Prunes environments from the registry that no longer exist on disk.
-    fn prune_nonexistent(&mut self) {
+    fn prune_nonexistent(&mut self, flox: &Flox) -> Result<(), EnvRegistryError> {
+        self.entries
+            .iter()
+            .filter(|entry| !entry.exists())
+            .try_for_each(|entry| {
+                for env in entry.envs.iter() {
+                    // Prune floxmeta branches for managed environments
+                    if let EnvironmentPointer::Managed(ref pointer) = env.pointer {
+                        let floxmeta = FloxMeta::open(flox, pointer)?;
+                        floxmeta.prune_branches(pointer, &entry.path)?;
+                    }
+                }
+
+                Ok(())
+            })
+            .map_err(EnvRegistryError::FloxMeta)?;
+
+        // Only mutate after successfully pruning branches.
```

Comment:
```suggestion
        // The environment registry is the only method we have of determining whether a branch in floxmeta should be garbage collected, so only remove entries after pruning floxmeta
```

Reply:
Related followup note: it seems like deleting from floxmeta should be a part of `deregister` rather than a separate step. For `ManagedEnvironment::delete` those are two separate steps